components/movementsensor/merged/merged.go

```diff
@@ -0,0 +1,373 @@
+	}
+	return m.linVel.LinearVelocity(ctx, extra)
+}
+
+func (m *merged) AngularVelocity(ctx context.Context, extra map[string]interface{}) (spatialmath.AngularVelocity, error) {
+	if m.angVel == nil {
+		return spatialmath.AngularVelocity{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedAngularVelocity
+	}
+	return m.angVel.AngularVelocity(ctx, extra)
+}
+
+func (m *merged) LinearAcceleration(ctx context.Context, extra map[string]interface{}) (r3.Vector, error) {
+	if m.linAcc == nil {
+		return r3.Vector{X: math.NaN(), Y: math.NaN(), Z: math.NaN()},
+			movementsensor.ErrMethodUnimplementedLinearAcceleration
+	}
+	return m.linAcc.LinearAcceleration(ctx, extra)
+}
+
+func (m *merged) Accuracy(ctx context.Context, extra map[string]interface{}) (map[string]float32, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	accMap := make(map[string]float32)
+
+	if m.ori != nil {
+		oriAcc, err := m.ori.Accuracy(ctx, extra)
+		if err != nil {
+			oriAcc = map[string]float32{"err_orienation_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(oriAcc, accMap)
+	}
+
+	if m.pos != nil {
+		posAcc, err := m.pos.Accuracy(ctx, extra)
+		if err != nil {
+			posAcc = map[string]float32{"err_position_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(posAcc, accMap)
+	}
+
+	if m.compass != nil {
+		compassAcc, err := m.compass.Accuracy(ctx, extra)
+		if err != nil {
+			compassAcc = map[string]float32{"err_compass_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(compassAcc, accMap)
+	}
+
+	if m.linVel != nil {
+		linvelAcc, err := m.linVel.Accuracy(ctx, extra)
+		if err != nil {
+			linvelAcc = map[string]float32{"err_linear_velocity_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(accMap, linvelAcc)
+	}
+
+	if m.angVel != nil {
+		angvelAcc, err := m.angVel.Accuracy(ctx, extra)
+		if err != nil {
+			angvelAcc = map[string]float32{"err_angular_velocity_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(accMap, angvelAcc)
+	}
+
+	if m.linAcc != nil {
+		linAccAcc, err := m.linAcc.Accuracy(ctx, extra)
+		if err != nil {
+			linAccAcc = map[string]float32{"err_angular_velocity_accuracy": float32(math.NaN())}
+		}
+		maps.Copy(linAccAcc, accMap)
+	}
+
+	return accMap, nil
+}
+
+func (m *merged) Properties(ctx context.Context, extra map[string]interface{}) (*movementsensor.Properties, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	return &movementsensor.Properties{
+		OrientationSupported:        m.oriReporting,
+		PositionSupported:           m.posReporting,
+		CompassHeadingSupported:     m.compassReporting,
+		LinearVelocitySupported:     m.linVelReporting,
+		AngularVelocitySupported:    m.angVelReporting,
+		LinearAccelerationSupported: m.linAccReporting,
+	}, nil
+}
+
+func (m *merged) Readings(ctx context.Context, extra map[string]interface{}) (map[string]interface{}, error) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	return movementsensor.Readings(ctx, m, extra)
+}
+
+func (m *merged) Close(context.Context) error {
+	return nil
```

Comment:
I think this is correct, but surprising. Put in a comment explaining that all the other sensors we're merging should not be closed just because we're closing.